### PR TITLE
Remove without_storage_info for the staking pallet

### PR DESCRIPTION
**Pull request by rajesh-nodle**
Part of [#173](https://github.com/NodleCode/chain-workspace/issues/173)

- Code Re-factor on Sub-functionalities ...
   -  Validator  :: Done
   -  Nominator  :: Done
   -  Slashing :: Done
   -  Hook :: Done

- Hooks Error Handling needs to be re-visited.

- Existing Unit
   -  2 test cases are verified, Have to move the remaining.

- Unit Test code extention
  - Add test cases to verify the new events & error codes :: TODO
 
**Status on Support Ticket raised to Parity**

- [11615](https://github.com/paritytech/substrate/issues/11615) - Closed

### CODE REVIEW

File: pallets/staking/src/hooks.rs [hunk 1]
```diff
@@ -119,18 +125,28 @@ impl<T: Config> pallet_session::SessionManager<T::AccountId> for Pallet<T> {
...
 
 		<BondedSessions<T>>::mutate(|bonded| {
-			bonded.push(start_index);
+			let _ = match bonded.try_push(start_index) {
+				Err(_) => {
+					log::error!(
+						"start_session:[{:#?}] - Error, Could be BondedSessions Overflow",
+						line!()
+					);
+					return ();
```
Comment (simonsso) [inline pallets/staking/src/hooks.rs:134]: Is this return needed here?

File: pallets/staking/src/hooks.rs [hunk 2]
```diff
@@ -119,18 +125,28 @@ impl<T: Config> pallet_session::SessionManager<T::AccountId> for Pallet<T> {
...
 
 				// prune out everything that's from before the first-kept index.
 				let n_to_prune = bonded
+					.to_vec()
 					.iter()
 					.take_while(|&&session_idx| session_idx < first_kept)
 					.count();
 
-				for prune_session in bonded.drain(..n_to_prune) {
+				for prune_session in bonded.to_vec().drain(..n_to_prune) {
```
Comment (simonsso) [inline pallets/staking/src/hooks.rs:149]: can't we get an iter from an bounded vec?

**Comment by rajesh-nodle** [comment]
Thanks for comment, pallet-staking is low priority right now. PR is not ready for review. will take up the comment when working on it.

**Comment by aliXsed** [comment]
@rajesh-nodle can you create move the rajesh-nodle:shamb0/bounded_staking to NodleCode/feature/bounded_staking so we keep the changes in that branch but close this PR here?

**Pull request closed without merging**